### PR TITLE
[docs] [orc8r] Add docs for module-module dependencies

### PR DESCRIPTION
**Pull request by AndyLKhuu**
Signed-off-by: Andy Lee Khuu <andykhuu@stanford.edu>

## Summary
This PR adds documentation for the high level dependencies that modules have on each other within the code base.
Screen Shot 2021-03-18 at 2 58 10 PM

## Test Plan
- [x] ./create_docusaurus.sh -> Manually Checking Local Host
- [x] cd magma/orc8r/cloud/docker ./build.py -t

 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by uri200** [comment]
Hey @AndyLKhuu , last week we introduced for first time FEG dependencies on LTE. And I see them on the picture.

Didn't you see some LTE references going into the FEG? For clarity it may make no sense to add that to this graph, but I am just curious.

This is the change that introduced FEG references into LTE and the reason why:

https://github.com/magma/magma/pull/5320#issuecomment-793289376

**Comment by AndyLKhuu** [comment]
> Hey @AndyLKhuu , last week we introduced for first time FEG dependencies on LTE. And I see them on the picture.
> 
> Didn't you see some LTE references going into the FEG? For clarity it may make no sense to add that to this graph, but I am just curious.
> 
> This is the change that introduced FEG references into LTE and the reason why:
> 
> [#5320 (comment)](https://github.com/magma/magma/pull/5320#issuecomment-793289376)

Hi Oriol, this graph only covers module to module dependencies from generated objects/definitions. Sorry for the confusion, will update changes to better reflect this point

File: docs/readmes/orc8r/dev_dependencies.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+id: dev_dependencies
+title: Module Dependencies
+hide_title: true
+---
+
+# Module dependencies on generated constructs
+
+Within Magma, each module(orc8r, lte, cwf, feg, wifi, fbinternal),
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_dependencies.md]: Missing a space before the open parenthesis

File: docs/readmes/orc8r/dev_dependencies.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+is in charge of generating and defining their own constructs. A module's
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_dependencies.md]: "constructs" -> "API objects, specifically defined as Swagger objects"

File: docs/readmes/orc8r/dev_dependencies.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+constructs may have dependencies on those of another module.
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_dependencies.md]: "A module's Swagger objects may depend on those from other modules. This results in the following module dependency tree"

**Review by hcgatewood — APPROVED** [review]
Lgtm, fix nits and msg me to merge

**Review by themarwhal — APPROVED** [review]
lgtm!

**Review by uri200 — APPROVED** [review]

File: docs/readmes/orc8r/dev_dependencies.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+id: dev_dependencies
+title: Module Dependencies
+hide_title: true
+---
+
+# Module dependencies on generated constructs
+
+Within Magma, each module (orc8r, lte, cwf, feg, wifi, fbinternal),
```
Comment (hcgatewood) [inline docs/readmes/orc8r/dev_dependencies.md]: Remove trailing comma after the closing parenthesis

**Review by hcgatewood — APPROVED** [review]
Saw one more small fix, then let's merge

**Pull request merged**